lib/queue.js

```diff
@@ -25,276 +48,197 @@ var FetchQueue = function() {
 module.exports = FetchQueue;
 
 FetchQueue.prototype = [];
-FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callback) {
+FetchQueue.prototype.add = function(queueItem, force, callback) {
+    callback = callback instanceof Function ? callback : noop;
+    var queue = this;
 
-    // For legacy reasons
-    if (depth instanceof Function) {
-        callback = depth;
-        depth = 1;
+    if (arguments.length === 2) {
```

Comment:
I wonder if we should leave this function with a pseudo-variadic calling style, or just make people send `force`. I'm leaning towards the latter, since we already provide a much friendlier queueing method on the crawler itself. But happy to hear arguments to the contrary!